lecture_3/Exercise_3/main.c

```diff
@@ -0,0 +1,125 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct node 
+{
+    int val;
+    struct node * next;
+} node_t;
+
+
+void listAdd(const node_t * head, int val) 
+{
+    node_t * current = head++;
+    while (current->next != NULL) 
+    {
+        current = current->next;
+    }
+
+    /* now we can add a new variable */
+    current->next = malloc(sizeof(node_t));
+    current->next->val = val;
+    current->next->next = NULL;
+}
+
+void listPrint(const node_t * head) 
+{
+    node_t * current = head;
+
+    while (current != NULL) 
+    {
+        printf("%d\n", current->val);
+        current = current->next;
+    }
+}
+
+int listRemoveLast(node_t * head) {
```

Comment:
I would prefer to have a function that allows to remove some item in the middle of a list, not just last element.

Reply:
Yes,  I first time added this function but when i was doing review this code i deleted one.